src/main/java/org/apache/nifi/documentation/xml/XmlDocumentationWriter.java

```diff
@@ -108,12 +108,12 @@ protected void writeDeprecationNotice(final DeprecationNotice deprecationNotice)
             return;
         }
 
-        final Class[] classes = deprecationNotice.alternatives();
+        final Class<?>[] classes = deprecationNotice.alternatives();
         final String[] classNames = deprecationNotice.classNames();
 
         final Set<String> alternatives = new LinkedHashSet<>();
         if (classes != null) {
```

Comment:
Is this `if` necessary if the default value in the `DeprecationNotice` is  `{}` which is an empty array and not null? Is there some way to specify `classNames`  in the `DeprecationNotice` and it for to be null? Same question for `classNames`.

Reply:
Thanks for highlighting that fact, it looks like the default value cannot be set as null, so this check can be removed.